lib/plugins/utils/isExtendedJson.js

```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 Ifiok Idiang.  All rights reserved.
+
+'use strict';
+
+/**
+ * Return the stripped content type string if it is a valid JSON extended type
+ *
+ * @public
+ * @function  validateExtendedJsonContentType
+ * @param   {String} contentType - the content type string to validate
+ * @returns {Boolean}       validity flag
+ */
+function validateExtendedJsonContentType(contentType) {
+    var type = contentType.toLowerCase();
+    // map any +json to application/json
+    var jsonPatternMatcher = new RegExp('^application/[a-zA-Z.]+\\+json');
```

Comment:
Interesting to see this RegExp being rebuilt on every call to the function. I built a little test that ran this function with the RegExp declared outside the function and found it to be about 460% faster. Is there a reason to leave it declared in the function rather than as a module var?

Reply:
Here is the test script if interested: ```
console.time('RegExp outside');
var jsonPatternMatcher = new RegExp('^application/[a-zA-Z.]+\\+json');
for (var i = 0; i < 100000; i++ ) {
  jsonPatternMatcher.test('application/json');
}
console.timeEnd('RegExp outside');

console.time('RegExp inside');
for (var i = 0; i < 100000; i++ ) {
  var jsonPatternMatcher = new RegExp('^application/[a-zA-Z.]+\\+json');
  jsonPatternMatcher.test('application/json');
}
console.timeEnd('RegExp inside');
```